rholang/src/main/bnfc/rholang_mercury.cf

```diff
@@ -101,20 +122,24 @@ LinearSimple. ReceiptLinearImpl ::= [LinearBind] ;
 -- LinearCond.   Linear ::= [LinearBind] "if" Proc ;
 
 -- Single Linear Bind
-LinearBindImpl. LinearBind ::= [Name] NameRemainder "<-" Name ;
-separator nonempty LinearBind ";" ;
+LinearBindImpl. LinearBind ::= [Name] NameRemainder "<-" NameSource ;
+separator nonempty LinearBind "&" ;
+
+SimpleSource.         NameSource ::= Name ;
+ReceiveSendSource.    NameSource ::= Name "?!" ;
+SendReceiveSource.    NameSource ::= Name "!?" "(" [Proc] ")" ;
 
 -- Repeated Receipts
 RepeatedSimple. ReceiptRepeatedImpl ::= [RepeatedBind] ;
 -- Single Repeated Bind
 RepeatedBindImpl. RepeatedBind ::= [Name] NameRemainder "<=" Name ;
-separator nonempty RepeatedBind ";" ;
+separator nonempty RepeatedBind "&" ;
```

Comment:
Question:
The rule changed here requires you to change the join ";" to "&" in `MakeMint.rho` and etc., right?

This looks like is not backward compatible from my perspective.Is that correct?

Reply:
Yes. That's correct. No, it's not backward compatible.